src/player_activity.cpp

```diff
@@ -473,3 +475,83 @@ void player_activity::finish( player *p )
         }
     }
 }
+
+template <typename T>
+bool containers_equal( const T &left, const T &right )
+{
+    if( left.size() != right.size() ) {
+        return false;
+    }
+
+    return std::equal( left.begin(), left.end(), right.begin() );
+}
+
+bool player_activity::can_resume_with( const player_activity &other, const Character &who ) const
+{
+    // Should be used for relative positions
+    // And to forbid resuming now-invalid crafting
+    ( void )who;
+    switch( type ) {
+        case ACT_WAIT:
+        case ACT_WAIT_NPC:
+        case ACT_WAIT_WEATHER:
+        case ACT_ARMOR_LAYERS:
+        case ACT_PULP:
+        case ACT_AIM:
+        case ACT_LONGSALVAGE:
+        case ACT_ATM:
+        case ACT_DROP:
+        case ACT_STASH:
+        case ACT_PICKUP:
+        case ACT_MOVE_ITEMS:
+        case ACT_ADV_INVENTORY:
+        case ACT_START_FIRE:
+        case ACT_FILL_LIQUID:
+            // Those shouldn't be resumed
+            // They don't store their progress in moves
+            return false;
+        case ACT_CRAFT:
+        case ACT_LONGCRAFT:
+            // Batch size is stored in values
+            // Coords may be matched incorrectly in some rare cases
+            // But it will not result in a false negative
+            if( !containers_equal( values, other.values ) ||
+                !containers_equal( coords, other.coords ) ) {
+                return false;
+            }
+
+            break;
+        case ACT_DISASSEMBLE:
+        // Disassembling is currently hardcoded in such a way
+        // that easy resuming isn't possible
+        case ACT_FORAGE:
+        case ACT_OPEN_GATE:
+        case ACT_OXYTORCH:
+        case ACT_CRACKING:
+        case ACT_FISH:
+        case ACT_PICKAXE:
+        case ACT_BURROW:
+        // Those should check position
+        // But position isn't set here yet!
+        // @todo Update the functions that set those activities
+        case ACT_BUTCHER:
+        case ACT_TRAIN:
+        case ACT_HOTWIRE_CAR:
+        case ACT_START_ENGINES:
+        case ACT_BUILD:
+        case ACT_VEHICLE:
+        case ACT_FIRSTAID:
+        case ACT_VIBE:
+        case ACT_MAKE_ZLAVE:
+        case ACT_GUNMOD_ADD:
+        case ACT_REPAIR_ITEM:
+            // Those should have extra limitations
+            // But for now it's better to allow too much than too little
+            break;
+        default:
```

Comment:
The default case prevents the compiler warning about missing enumeration values. Is it really necessary, or can you just add the missing values as additional cases (instead of the default case)?